src/sif/autocost.cc

```diff
@@ -295,10 +314,11 @@ class AutoCost : public DynamicCost {
   virtual bool Allowed(const baldr::DirectedEdge* edge,
                        const graph_tile_ptr& tile,
                        uint16_t disallow_mask = kDisallowNone) const override {
+    bool allow_hov = IsHOVAllowed(edge);
```

Comment:
why do this? let the logical  expression below shortcircuit and never even call this function if it doesnt need to, ie just put the function call at the end of the return statement

Reply:
My preference is to write code that is readable and debuggable.  By breaking out the `bool allow_hov` I can set a breakpoint after it is initialized to inspect its value.  The optimizer can optimize it away in release builds.  That my reasoning.  But I can change it.